src/bindgen/language_backend/java_jna.rs

```diff
@@ -159,13 +160,20 @@ impl LanguageBackend for JavaJnaLanguageBackend<'_> {
     fn write_footers<W: Write>(&mut self, _: &mut SourceWriter<W>) {}
 
     fn write_enum<W: Write>(&mut self, out: &mut SourceWriter<W>, e: &Enum) {
+        let has_data = e.tag.is_some();
+
+        // First, write the enum as an integer type
+        let mut tag_name = e.export_name.clone();
+        if has_data {
+            tag_name = format!("{}Tag", e.export_name);
+        }
+
         self.write_integer_type(
             out,
             &JnaIntegerType {
                 documentation: &e.documentation,
-                name: &e.export_name,
-                /* enum are most of the time the same size as ints */
-                underlying_jna_integer_type: UnderlyingJnaIntegerType::Int,
+                name: &tag_name,
+                underlying_jna_integer_type: UnderlyingJnaIntegerType::Long,
```

Comment:
hum not sure this is correct

Reply:
You can take a look at this; https://www.eventhelix.com/rust/rust-to-assembly-enum-match/#:~:text=The%20size%20of%20the%20enum%20is%2024%20bytes.,currently%20saved%20in%20the%20enum.&text=Notes%3A,with%20%23%5Brepr(C)%5D%20.

Also, if you have a typed enum like I do, you will notice that the field alignments of the union fields is wrong with a tag of size 4 bytes. It needs to be packed into 8 bytes.

Reply:
Ah I see, yeah for some reason JNA wasn't aligning my typed variants. For eg. if I had an enum with a variant with a 32 bytes array and the first 4 bytes were the rest of the enum variant idx, since it was 8 bytes. Maybe there is an option in JNA to handle packing?